build.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+# kvikio build script
+
+# This script is used to build the component(s) in this repo from
+# source, and can be called with various options to customize the
+# build as needed (see the help output for details)
+
+# Abort script on first error
+set -e
+
+NUMARGS=$#
+ARGS=$*
+
+# NOTE: ensure all dir changes are relative to the location of this
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean libkvikio kvikio -v -g -n -s --ptds -h"
+HELP="$0 [clean] [libkvikio] [kvikio] [-v] [-g] [-n] [-s] [--ptds] [--cmake-args=\"<args>\"] [-h]
+   clean                       - remove all existing build artifacts and configuration (start over)
+   libkvikio                   - build and install the libkvikio C++ code
+   kvikio                      - build and install the kvikio Python package
+   -v                          - verbose build mode
+   -g                          - build for debug
+   -n                          - no install step
+   -s                          - statically link against cudart
+   --ptds                      - enable per-thread default stream
+   --cmake-args=\\\"<args>\\\" - pass arbitrary list of CMake configuration options (escape all quotes in argument)
+   -h                          - print this text
+   default action (no args) is to build and install 'libkvikio' and 'kvikio' targets
+"
+LIBKVIKIO_BUILD_DIR=${LIBKVIKIO_BUILD_DIR:=${REPODIR}/cpp/build}
+KVIKIO_BUILD_DIR="${REPODIR}/python/build ${REPODIR}/python/_skbuild"
+BUILD_DIRS="${LIBKVIKIO_BUILD_DIR} ${KVIKIO_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE_FLAG=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+BUILD_BENCHMARKS=OFF
+BUILD_TESTS=OFF
+CUDA_STATIC_RUNTIME=OFF
+PER_THREAD_DEFAULT_STREAM=OFF
```

Comment:
Some of these variables may be unused. I don't see any usage of `BUILD_BENCHMARKS`, `BUILD_TESTS`, `CUDA_STATIC_RUNTIME`, or `PER_THREAD_DEFAULT_STREAM` aside from accepting the arguments?